python/graphstorm/inference/emb_infer.py

```diff
@@ -0,0 +1,144 @@
+            assert len(loader.data.eval_ntypes) == 1, \
+                "GraphStorm only support single target node type for training and inference"
+
+        assert save_embed_path is not None
+
+        sys_tracker.check('start embedding generation')
+        self._model.eval()
+
+        if task_type == BUILTIN_TASK_LINK_PREDICTION:
+            # for embedding generation, it is preferred to use full graph
+            if use_mini_batch_infer:
+                embs = do_mini_batch_inference(self._model, data, fanout=loader.fanout,
+                                               edge_mask=None,
+                                               task_tracker=self.task_tracker)
+            else:
+                embs = do_full_graph_inference(self._model, data, fanout=loader.fanout,
+                                               edge_mask=None,
+                                               task_tracker=self.task_tracker)
+        elif task_type in {BUILTIN_TASK_NODE_REGRESSION, BUILTIN_TASK_NODE_CLASSIFICATION}:
+            # only generate embeddings on the target node type
+            ntype = loader.data.eval_ntypes[0]
+            if use_mini_batch_infer:
+                inter_embs = node_mini_batch_gnn_predict(self._model, loader, return_proba,
+                                                  return_label=False)[1]
+                inter_embs = {ntype: inter_embs[ntype]} if isinstance(inter_embs, dict) \
+                    else {ntype: inter_embs}
+                g = loader.data.g
+                ntype_emb = create_dist_tensor((g.num_nodes(ntype), inter_embs[ntype].shape[1]),
+                                               dtype=inter_embs[ntype].dtype,
+                                               name=f'gen-emb-{ntype}',
+                                               part_policy=g.get_node_partition_policy(ntype),
+                                               persistent=True)
+                ntype_emb[loader.target_nidx[ntype]] = inter_embs[ntype]
+                embs = {ntype: ntype_emb}
+            else:
+                embs = do_full_graph_inference(self._model, data, fanout=loader.fanout,
+                                               task_tracker=self.task_tracker)
+                ntype_emb = embs[ntype]
+                embs = {ntype: ntype_emb}
+        elif task_type in {BUILTIN_TASK_EDGE_CLASSIFICATION, BUILTIN_TASK_EDGE_REGRESSION}:
+            # Currently it is not allowed to do mini-batch inference
+            # and save embedding on edge tasks
+            embs = do_full_graph_inference(self._model, loader.data, fanout=loader.fanout,
+                                           task_tracker=self.task_tracker)
+            target_ntypes = set()
+            for etype in loader.data.eval_etypes:
+                target_ntypes.add(etype[0])
+                target_ntypes.add(etype[2])
+
+            embs = {ntype: embs[ntype] for ntype in sorted(target_ntypes)}
```

Comment:
For mini-batch inference, we just iterate all target_ntypes and compute node embeddings for each ntype.